mongodump/mongodump_test.go

```diff
@@ -579,9 +587,47 @@ func TestMongoDumpBSON(t *testing.T) {
 					md.OutputOptions.Archive = "dump_slash.archive"
 					err = md.Init()
 					So(err, ShouldBeNil)
+				})
+			})
+
+			Convey("and that it dumps a collection with a name >238 bytes in the right format", func() {
```

Comment:
This should only be run on >4.4. We'll also need to add a 4.4 task to evergreen. As long as this wrks locally on 4.4, we can add the 4.4 task in another PR. But do make sure to add the 4.4 check on this test now.